### PR TITLE
Mvvm2

### PR DESCRIPTION
**Pull request by tbuschto**

### CODE REVIEW

File: src/decorators/bind.ts [hunk 1]
```diff
@@ -170,7 +170,7 @@ function scheduleIsComponentCheck(binding: BindSuperConfig<unknown>) {
     try {
       checkIsComponent(binding.componentProto);
     } catch (ex) {
-      const target = binding.all ? JSON.stringify(binding.all) : binding.targetPath.join('.');
+      const target = binding.all ? JSON.stringify(binding.all) : binding.targetPath.slice(1).join('.');
```
Comment (cpetrov) [inline src/decorators/bind.ts:173]: My initial expectation would be that the binding path would not include the direction of the binding? Would this be easier to understand if this property had another name, e.g. `binding.spec` or something like that?

File: doc/databinding/@bindAll.md [hunk 2]
```diff
@@ -93,14 +95,12 @@ class Model {
...
 
-If a target property converts or ignores the incoming value of a source property, the source property will ignore that and keep its own value. The two properties are out-of-sync in this case.
+In `model << child` bindings the model property receiving the data will always be set to the value of the child property sending the data, whether it previously was `undefined` or some other value.
 
-If either property throws when set, the error will be propagated to the caller that originally caused the value change. In this case the two properties *may* end up out-of-sync.
+If in a two-way binding a source (model) property converts or ignores the incoming value of the target property, the target property will follow and also bet set to the new source property value. If a target property converts or ignores the incoming value of a source property, the source property will ignore that and keep its own value. The two properties are out-of-sync in this case. If either property throws when set, the error will be propagated to the caller that originally caused the value change. In this case the two properties *may* end up out-of-sync.
```
[the remark this answers is not in the document]
Comment (tbuschto) [reply doc/databinding/@bindAll.md:106]: done

File: src/decorators/bind.ts (hunk 1 above)
Comment (tbuschto) [reply src/decorators/bind.ts:173]: can't really change that without completely re-doing the API

File: doc/databinding/@bindAll.md [hunk 3]
```diff
@@ -93,14 +95,12 @@ class Model {
...
 
-There is one exception to this behavior: If the source property is set to `undefined` (but not `null`) at that moment it will be assigned the current value of the target property. Likewise, if a source property is set to `undefined` after the two-way binding has been established, both properties will be set to the initial value of the target property.
+When a new two-way or `model >> child` binding is established (when `append` is called or the component property is assigned a new source object) all the target properties will be set to the current value of the their respective source property.
 
-If a source property converts or ignores the incoming value of the target property, the target property will follow and also bet set to the new source property value.
+There are two exceptions to this behavior: If the source property is set to `undefined` (but not `null`) at that moment it will be assigned the current value of the target property. Likewise, if a source property is set to `undefined` after the two-way binding has been established, both properties will be set to the initial value of the target property.
```
[the remark this answers is not in the document]
Comment (tbuschto) [reply doc/databinding/@bindAll.md:102]: done

**Pull request closed without merging**